livre/core/pipeline/FilterOutput.h

```diff
@@ -0,0 +1,202 @@
+                                                      + portName ));
+
+        return getPort( portName )->getValue< T >();
+    }
+
+    /**
+     * Moves the value with the given type T
+     * @param portName is the name of the port.
+     * @return the value.
+     * @throw std::runtime_error whem the port with the port name
+     * is not available.
+     * @throw std::runtime_error when the port data is not exact
+     * type T
+     */
+    template< class T >
+    const T&& moveValue( const std::string& portName ) const
+    {
+        if(!hasPort( portName ))
+            LBTHROW( std::runtime_error( std::string( "No output port with name: ")
+                                                      + portName ));
+
+        return getPort( portName )->moveValue< T >();
+    }
+
+    /**
+     * Sets the value with the given type T
+     * @param portName is the name of the port.
+     * @return the value reference.
+     * @throw std::runtime_error whem the port with the port name
+     * is not available.
+     * @throw std::runtime_error when the port data is not exact
+     * type T
+     */
+    template< class T >
+    void setValue( const std::string& portName,
+                   const T& value )
+    {
+        if(!hasPort( portName ))
+            LBTHROW( std::runtime_error( std::string( "No output port with name: ")
+                                                      + portName ));
+
+        return getPort( portName )->setValue( value );
+    }
+
+    /**
+     * @return the list of port names.
+     */
+    lunchbox::Strings getPortNames() const;
+
+    /**
+     * Waits for a given port name or for all the output ports if
+     * portname is empty.
+     * @param portName is the name of the port. Empty for all ports.
+     * @throw std::runtime_error whem the port with the port name
+     * is not available.
+     */
+    void wait( const std::string& portName = "" ) const;
+
+    /**
+     * Waits for a any input to be ready.
+     * @param readyPorts The ready ports are listed into readyPorts array.
+     * @return true there are still waiting outputs.
+     */
+    bool waitForAny( lunchbox::Strings& readyPorts ) const;
+
+    /**
+     * Queries given port name or for all the input ports if
+     * port is ready.
+     * @param portName is the name of the port. Empty for all ports.
+     * @return true if port/ports are ready.
+     * is not available.
+     * @throw std::runtime_error whem the port with the port name
+     */
+    bool isReady( const std::string& portName = "" ) const;
+
+    /**
+     * Connect given pipe filters with the given port name. Both filters
+     * should have the same port name and their types should match.
+     * @param src is the source pipe filter.
+     * @param dst is the destination pipe filter.
+     * @param portName connection port name.
+     * @return true if connection is successful.
+     */
+    friend bool connect( PipeFilterPtr& src,
```

Comment:
Non const refs on ptr types? Is that correct? If the function modified the ptrs then it should be documented